### PR TITLE
New features for offline Tk DQM scripts: 76X

### PR DESCRIPTION
**Pull request by vdutta**
76X version of #10757
- Automatic creation of root files with information from SiStrip tracker maps indexed by DetId, will be done at the time of tracker map creation by Tracker Offline shifter.
- Addition of run numbers to the headings of the tracker maps created offline.

### CODE REVIEW

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
+1
for https://github.com/vdutta/cmssw/commit/693b5b9e3793ffa9794c833df329b5102da382f2 provided jenkins agrees

**Comment by deguio** [comment]
ciao @vdutta 
I am scared that the tree writing (false by default) can become true (by mistake) in production at some point. those modules run at the T0.. it is not about scripts...
F.

**Comment by threus** [comment]
@deguio I see what you mean, but the only overwriting of the flag that is used to check whether a root file is produced happens in the script that is not run at T0 (so it's false by default in all official sequences). I understand that rewrites during testing might end up unintentionally in future PRs, I am not sure how we can avoid using flags. Do you have any suggestion?

**Comment by deguio** [comment]
-1
in the meantime this PR is not mergeable anymore..

I don't have smart ideas other than duplicating the code... I guess that the PR can be merged as soon as it is rebased.

sorry for the late reply,
F.

**Comment by deguio** [comment]
please test

**Comment by deguio** [comment]
+1

**Comment by deguio** [comment]
please test

**Comment by deguio** [comment]
+1

**Comment by mmusich** [comment]
+1

File: CalibTracker/SiStripQuality/plugins/SiStripQualityStatistics.cc [hunk 1]
```diff
@@ -34,11 +34,15 @@ SiStripQualityStatistics::SiStripQualityStatistics( const edm::ParameterSet& iCo
   TkMapFileName_(iConfig.getUntrackedParameter<std::string>("TkMapFileName","")),
   fp_(iConfig.getUntrackedParameter<edm::FileInPath>("file",edm::FileInPath("CalibTracker/SiStripCommon/data/SiStripDetInfo.dat"))),
   saveTkHistoMap_(iConfig.getUntrackedParameter<bool>("SaveTkHistoMap",true)),
+  runNumberA_(iConfig.getUntrackedParameter<unsigned long long>("RunNumber",0)),
```
Comment (davidlange6) [inline CalibTracker/SiStripQuality/plugins/SiStripQualityStatistics.cc]: @vdutta 
run numbers in CMS are an unsigned int  (0-4 million something). Why a unsigned long long?

Comment (Dr15Jones) [inline CalibTracker/SiStripQuality/plugins/SiStripQualityStatistics.cc]: Just to be pedantic, the first valid run number is 1.

**Comment by deguio** [comment]
@vdutta do you think you are going to update this based on the comments you got?
thanks,
F.

**Comment by vdutta** [comment]
Hi @deguio, the addition of the run numbers was actually from someone else, but having a look it should not be an issue to update this based on the comments. I can do this for this PR and #10758 (75X), but for #10757 (74X) I was worried this would delay the integration further, do we also need the changes there?

**Comment by deguio** [comment]
ciao @vdutta 
74x will be abandoned soon. adding the change to 75 and 76 and 80 should be enough.
thanks,
F.

**Comment by boudoul** [comment]
ciao , yes would be good to have it at least for the next HI datataking .... so 75X (for HI), and then 76X 80X to keep eveything synchronzied , since we couldn' get in for the current 74X pp data taking ...

**Comment by boudoul** [comment]
btw @vdutta, could you please make a clear description of what this PR is doing in the description box? thanks

**Comment by vdutta** [comment]
@boudoul Done, please let me know if it's unclear.

**Comment by deguio** [comment]
+1

**Comment by deguio** [comment]
please test

**Comment by deguio** [comment]
please test

**Comment by threus** [comment]
hi all, is there still something that prevents this PR to be finally merged? Thanks.

**Comment by mmusich** [comment]
+1

**Comment by davidlange6** [comment]
I am cleaning up the 76x queue aside from things for analysis workflows. I'm closing this pull request, please make sure the PR is in 80x. Thanks!

**Pull request closed without merging**